pkg/quota_management/account.go

```diff
@@ -1,20 +1,69 @@
 package quota_management
 
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+)
+
 type Account struct {
-	Username            string `yaml:"username"`
-	MaxAllowedInstances int    `yaml:"max_allowed_instances"`
+	Username            string    `yaml:"username"`
+	MaxAllowedInstances int       `yaml:"max_allowed_instances"`
+	GrantedQuota        QuotaList `yaml:"granted_quota,omitempty"`
 }
 
-func (account Account) IsInstanceCountWithinLimit(count int) bool {
-	return count <= account.GetMaxAllowedInstances()
+var _ QuotaManagementListItem = &Account{}
+
+func (account Account) IsInstanceCountWithinLimit(instanceTypeID string, billingModelName string, count int) bool {
+	if billingModelName == "" {
```

Comment:
Is there some reason we have a default?

I think we should consider avoiding implicit behavior around this, as with the AMS implementation it's proven to increase complexity substantially

Reply:
I totally agree. The idea here was to guarantee compatibility with the current clients. When the clients (UI and CLI) are updated, we can then remove the default.

Reply:
Shouldn't we be setting billingModel beforehand in this case?

Reply:
I moved it elsewhere (caller).